### DISCUSSION TITLE
new to next.js - reasons for pairing next with express

### DISCUSSION THREAD

**Discussion by jaybytez** [vercel/next.js#12299] [Help]
We are looking to build a site using next.js for static site and server side generation.  One of the internal suggestions is to run next with express on node.  Is there a reason you would pair next.js with express?  I haven't found any really good resources except one blog post from logrocket.  Thanks in advance.

**Comment by Timer** [comment]
Pairing Next.js with Express was only necessary before Next.js 9. It is no longer a recommended pattern!

It used to be useful for dynamic routing, but that feature is now built into Next.js.

Using Express with Next.js actually ends up reducing your application performance, and makes deployment more difficult (not compatible with any Jamstack platforms, e.g. Vercel, which Next.js is designed for).

**Reply by kamilml** [reply]
I have a question about [How to set httpOnly cookies? #12301](https://github.com/zeit/next.js/discussions/12301)

in old versions I was using express to make a reverse proxy, because my backend API is with another domain

what is the solution without express?
